zbench.zig

```diff
@@ -286,13 +300,54 @@ pub const Benchmark = struct {
                 progress_node.setEstimatedTotalItems(0);
                 progress_node.setCompletedItems(0);
                 progress.refresh();
+
+                // Sum the durations from the timings_ns array
+                var total_duration_ns: u64 = 0;
+                for (x.readings.timings_ns) |time_ns| {
+                    total_duration_ns += time_ns;
+                }
+
+                // Append the result with the total duration
+                const newResult = try Result.init(x.name, x.readings, total_duration_ns);
+                try results.append(newResult); // Append the successfully created Result
+
+                if (total_duration_ns < fastest) {
+                    fastest = total_duration_ns;
+                    fastest_name = x.name;
+                }
+
                 try x.prettyPrint(arena.allocator(), writer, true);
                 _ = arena.reset(.retain_capacity);
             },
         };
+
+        // Print summary
+        if (self.common_config.display_summary) {
+            try printSummary(writer, results.items, fastest, fastest_name);
+        }
     }
 };
 
+fn printSummary(writer: anytype, results: []const Result, fastest_ns: u64, fastest_name: []const u8) !void {
+    try writer.print("\x1b[1mSummary:\x1b[0m\n", .{});
+    try writer.print("{s} ran\n", .{fastest_name});
+
```

Comment:
It might be cleaner to work out which was the fastest benchmark in `printSummary` rather than adding that code to the `run` function, all you need is an array of the total durations and if the `Benchmark` is passed as well you can match those durations to the benchmark definitions by index to get their names. It might make sense as a method then if the `Benchmark` is a parameter.